pkg/store/driver/txn/unionstore_driver.go

```diff
@@ -195,6 +199,34 @@ func getTiDBKeyFlags(flag tikvstore.KeyFlags) kv.KeyFlags {
 	return v
 }
 
+type snapshot struct {
```

Comment:
Or just name it `MemBufferSnapshot` since there are many `snapshot` like definitions in TiDB?